actions/service.go

```diff
@@ -174,66 +183,101 @@ func (s *Service) runTasks(ctx context.Context, record graveler.HookRecord, task
 	return g.Wait().ErrorOrNil()
 }
 
-func (s *Service) insertRunInformation(ctx context.Context, record graveler.HookRecord, tasks [][]*Task) error {
+func (s *Service) saveRunInformation(ctx context.Context, record graveler.HookRecord, tasks [][]*Task) error {
 	if len(tasks) == 0 {
 		return nil
 	}
-	// collect run information based on tasks
-	runStartTime, runEndTime, runPassed := runInformationBasedOnTasks(tasks)
 
-	// save run and tasks information
+	manifest := buildRunManifest(record, tasks)
+
+	err := s.insertRunInformation(ctx, record.RepositoryID, manifest)
+	if err != nil {
+		return fmt.Errorf("insert run information: %w", err)
+	}
+
+	return s.saveRunManifest(ctx, record, manifest)
+}
+
+func (s *Service) saveRunManifest(ctx context.Context, record graveler.HookRecord, manifest runManifest) error {
+	manifestJSON, err := json.Marshal(manifest)
+	if err != nil {
+		return fmt.Errorf("marshal run manifest: %w", err)
+	}
+	runManifestPath := FormatRunManifestOutputPath(record.RunID)
+	manifestReader := bytes.NewReader(manifestJSON)
+	manifestSize := int64(len(manifestJSON))
+	storageNamespace := record.StorageNamespace.String()
+	return s.Writer.OutputWrite(ctx, storageNamespace, runManifestPath, manifestReader, manifestSize)
+}
+
+func (s *Service) insertRunInformation(ctx context.Context, repositoryID graveler.RepositoryID, manifest runManifest) error {
 	_, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
 		// insert run information
+		run := manifest.Run
 		_, err := tx.Exec(`INSERT INTO actions_runs(repository_id, run_id, event_type, start_time, end_time, branch_id, source_ref, commit_id, passed)
 			VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9)`,
-			record.RepositoryID, record.RunID, record.EventType, runStartTime, runEndTime, record.BranchID, record.SourceRef, record.CommitID, runPassed)
+			repositoryID, run.RunID, run.EventType, run.StartTime, run.EndTime, run.BranchID, run.SourceRef, run.CommitID, run.Passed)
 		if err != nil {
-			return nil, fmt.Errorf("insert run information: %w", err)
+			return nil, fmt.Errorf("insert run information %s: %w", run.RunID, err)
 		}
 
 		// insert each task information
-		for _, actionTasks := range tasks {
-			for _, task := range actionTasks {
-				if task.StartTime.IsZero() {
-					break
-				}
-				taskPassed := task.Err == nil
-				_, err = tx.Exec(`INSERT INTO actions_run_hooks(repository_id, run_id, hook_run_id, action_name, hook_id, start_time, end_time, passed)
-			VALUES ($1,$2,$3,$4,$5,$6,$7,$8)`,
-					record.RepositoryID, task.RunID, task.HookRunID, task.Action.Name, task.HookID, task.StartTime, task.EndTime, taskPassed)
-				if err != nil {
-					return nil, fmt.Errorf("insert run hook information (%s %s): %w", task.Action.Name, task.HookID, err)
-				}
+		for _, hookRun := range manifest.HooksRun {
```

Comment:
Just a thought (@ozkatz thought really): Can we avoid writing to the db and read directly from the os?
I think we can, the only thing missing is the ability to filter by branches or specific hook runs.

Reply:
will need need to update our adapters to support more options in the listing and filter, but possible.
opened https://github.com/treeverse/lakeFS/issues/1511